buildSrc/build.gradle

```diff
@@ -78,50 +89,80 @@ dependencies {
-task writeVersionProperties {
-  inputs.properties(props)
-  outputs.file(tempPropertiesFile)
-  doLast {
-    OutputStream stream = Files.newOutputStream(tempPropertiesFile.toPath());
-    try {
-      props.store(stream, "UTF-8");
-    } finally {
-      stream.close();
+
+/*****************************************************************************
+ *                    Bootstrap repositories and IDE setup                   *
+ *****************************************************************************/
+// this will only happen when buildSrc is built on its own during build init
+if (project == rootProject) {
+
+  // TODO: move common IDE configuration to a common file to include
+  apply plugin: 'idea'
+  apply plugin: 'eclipse'
+
+  repositories {
+    mavenCentral()
+    maven {
+      name 'sonatype-snapshots'
+      url "https://oss.sonatype.org/content/repositories/snapshots/"
     }
   }
-}
-
-processResources {
-  dependsOn writeVersionProperties
-  from tempPropertiesFile
-}
 
-extraArchive {
-  javadoc = false
-  tests = false
-}
+  idea {
+    module {
+      inheritOutputDirs = false
+      outputDir = file('build-idea/classes/main')
+      testOutputDir = file('build-idea/classes/test')
+    }
+  }
 
-idea {
-  module {
-    inheritOutputDirs = false
-    outputDir = file('build-idea/classes/main')
-    testOutputDir = file('build-idea/classes/test')
+  eclipse {
+    classpath {
+      defaultOutputDir = file('build-eclipse')
+    }
   }
-}
 
-eclipse {
-  classpath {
-    defaultOutputDir = file('build-eclipse')
+  task copyEclipseSettings(type: Copy) {
+    from project.file('src/main/resources/eclipse.settings')
+    into '.settings'
+  }
+  // otherwise .settings is not nuked entirely
+  tasks.cleanEclipse {
+    delete '.settings'
   }
+  tasks.eclipse.dependsOn(cleanEclipse, copyEclipseSettings)
 }
 
-task copyEclipseSettings(type: Copy) {
-  from project.file('src/main/resources/eclipse.settings')
-  into '.settings'
-}
-// otherwise .settings is not nuked entirely
-tasks.cleanEclipse {
-  delete '.settings'
+/*****************************************************************************
+ *                           Normal project checks                           *
+ *****************************************************************************/
+
+// this happens when included as a normal project in the build, which we do
+// to enforce precommit checks like forbidden apis, as well as setup publishing
+if (project != rootProject) {
+  apply plugin: 'elasticsearch.build'
+  apply plugin: 'nebula.maven-base-publish'
+  apply plugin: 'nebula.maven-scm'
+
+  // groovydoc succeeds, but has some weird internal exception...
+  groovydoc.enabled = false
+
+  // build-tools is not ready for primetime with these...
+  dependencyLicenses.enabled = false
+  forbiddenApisMain.enabled = false
+  jarHell.enabled = false
+  loggerUsageCheck.enabled = false
+  thirdPartyAudit.enabled = false
```

Comment:
Yeah, thirdPartyAudit is going to be fun. I can't imagine gradle passing. Maybe we just give a few of these a permanent pass because this is build code?